### ISSUE TITLE
Nested arguments break CLI with --flatten 

### ISSUE THREAD

**Issue by carrot-sticks** [formatjs/formatjs#3849]
## Which package?
`@formatjs/cli`

## Describe the bug

When the message contains nested interpolation arguments, running `extract` with `--flatten` will produce the following error:

```
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at cloneDeep (/.../node_modules/@formatjs/cli/bin/formatjs:154639:23)
    at /.../node_modules/@formatjs/cli/bin/formatjs:154640:23
    at Array.reduce (<anonymous>)
    at cloneDeep (/.../node_modules/@formatjs/cli/bin/formatjs:154639:33)
    at Array.map (<anonymous>)
    at cloneDeep (/.../node_modules/@formatjs/cli/bin/formatjs:154636:46)
    at /.../node_modules/@formatjs/cli/bin/formatjs:154640:23
    at Array.reduce (<anonymous>)
    at cloneDeep (/.../node_modules/@formatjs/cli/bin/formatjs:154639:33)
```

(I replace project path with `...` for privacy reasons.)

## To Reproduce

1. Put the following message in the `src/index.tsx`:

```tsx
        <FormattedMessage
          id="membersCount"
          defaultMessage="{count, plural, one {{count, number} member} other {{count, number} members}}"
          values={{ count: 5 }}
        />
```

2. Try to extract it with:

```
yarn run formatjs extract 'src/**/*.{ts,tsx}' --format simple --flatten --out-file en.json
```

## Expected behavior

Produce the following output file. 

```json 
{
  "membersCount": "{count,plural,one{{count, number} member} other{{count, number} members}}"
}
```

BDW, if I use `{count, number} {count, plural, one {member} other {members}}` as a `defaultMessage` I got the desired result, but that may not be possible for all translations.

**Comment by carrot-sticks** [comment]
I created the MR. Please, look at it. I know that I probably didn't follow your conventions, and I need help with this.

**Comment by longlho** [comment]
fixed in d1c78e9076bf8499b89a5870dfa3bdb5ead4a741
